hpx/runtime/serialization/variant.hpp

```diff
@@ -0,0 +1,78 @@
+//  Copyright (c) 2015 Anton Bikineev
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef HPX_SERIALIZATION_VARIANT_HPP
+#define HPX_SERIALIZATION_VARIANT_HPP
+
+#include <hpx/runtime/serialization/serialize.hpp>
+
+#include <boost/variant.hpp>
+
+namespace hpx { namespace serialization
+{
+    namespace detail
+    {
+        struct variant_save_visitor: boost::static_visitor<>
+        {
+            variant_save_visitor(output_archive& ar)
+              : ar(ar)
+            {}
+
+            template <class T>
+            void operator()(const T& value)
+            {
+                ar << value;
+            }
+
+        private:
+            output_archive& ar;
+        };
+
+        template <class... Args>
+        struct load_helper_t{};
+
+        template <class Variant, class... Tail>
+        BOOST_FORCEINLINE void load_helper(input_archive& ar, Variant& var,
+                int which, load_helper_t<boost::detail::variant::void_, Tail...>)
```

Comment:
They are actually the same as BOOST_PP_ENUM_PARAMS... macros where first parameter is defined as a limit of template parameters for variant. This is how serialization of variant is implemented inside boost.serialization (it's still recursive but with help of mpl::list). Currently I see two ways:
1) existing with variadic templates and explicit use of detail::variant::void_;
2) one used in boost.serialization with these macros expanded to class T1, ... class T50 and use of mpl::list for recursion.
I don't see a big problem with use of such implementation detail.